packages/account/src/Sections/Security/Passkeys/passkeys.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import { Loading } from '@deriv/components';
+import { useGetPasskeysList, useIsPasskeySupported, useRegisterPasskey } from '@deriv/hooks';
+import { routes } from '@deriv/shared';
+import { observer, useStore } from '@deriv/stores';
+import PasskeysStatusContainer from './components/passkeys-status-container';
+import PasskeysList from './components/passkeys-list';
+import PasskeyModal from './components/passkey-modal';
+import { getModalContent, PASSKEY_STATUS_CODES, TPasskeysStatus } from './passkeys-configs';
+import './passkeys.scss';
+
+const Passkeys = observer(() => {
+    const { ui } = useStore();
+    const { is_mobile } = ui;
+
+    const [passkey_status, setPasskeyStatus] = React.useState<TPasskeysStatus>(PASSKEY_STATUS_CODES.NONE);
+    const { is_passkey_supported, is_passkey_support_checking } = useIsPasskeySupported();
+    const { passkeys_list, is_passkeys_list_loading, passkeys_list_error, reloadPasskeysList } = useGetPasskeysList();
+    const {
+        cancelPasskeyRegistration,
+        createPasskey,
+        clearPasskeyRegistrationError,
+        startPasskeyRegistration,
+        is_passkey_registration_started,
+        is_passkey_registered,
+        passkey_registration_error,
+    } = useRegisterPasskey();
+
+    const should_show_passkeys = is_passkey_supported && is_mobile;
+
+    React.useEffect(() => {
+        if (!passkeys_list?.length && !is_passkey_registered) {
+            setPasskeyStatus(PASSKEY_STATUS_CODES.NO_PASSKEY);
+        } else if (is_passkey_registered) {
+            setPasskeyStatus(PASSKEY_STATUS_CODES.CREATED);
+        } else {
+            setPasskeyStatus(PASSKEY_STATUS_CODES.NONE);
+        }
+    }, [is_passkey_registered, passkeys_list]);
+
+    if (is_passkey_support_checking || is_passkeys_list_loading) {
+        return <Loading is_fullscreen={false} className='account__initial-loader' />;
+    }
+    if (!should_show_passkeys) {
+        return <Redirect to={routes.traders_hub} />;
+    }
+
+    const error = passkeys_list_error || passkey_registration_error;
+    const modal_content = getModalContent({
+        error,
+        is_passkey_registration_started,
+    });
+
+    const onCloseErrorModal = () => {
+        if (passkeys_list_error) {
+            reloadPasskeysList();
+        }
+        if (passkey_registration_error) {
+            clearPasskeyRegistrationError();
+        }
```

Comment:
will the 2 error happen at the same time? If no then you can use `if else` instead of 2 if

Reply:
in the same time - no, because they are different flows.
 I just wanted to be explicit here with the error check, 
so if I leave here the check for existence of passkey_registration_error, 'else' will be redundant, will it?
@suisin-deriv